WEB_SERVICE/WEB/_HIGH/WEAK_MAC_ALGORITHM/recommendation.md

```diff
@@ -0,0 +1,43 @@
+To mitigate the risks associated with weak MAC algorithms, consider the following recommendations:
+
+1. **Disable Weak MAC Algorithms:**
+   - Disable support for weak MAC algorithms such as MD5 and SHA1.
+   - Enable only strong MAC algorithms like SHA256, SHA384, or SHA512.
+
+2. **Configure Strong Cipher Suites:**
+   - Use cipher suites that incorporate strong MAC algorithms.
+   - Disable cipher suites that use weak MAC algorithms.
+
+3. **Implement Secure TLS Configuration:**
+   - Follow industry best practices for TLS configuration, such as those provided by Mozilla's SSL Configuration Generator or OWASP's TLS Cheat Sheet.
+   - Regularly test your TLS configuration using tools like SSL Labs' SSL Server Test.
+
+4. **Use HTTP Strict Transport Security (HSTS):**
+   - Implement HSTS to ensure that clients always connect to your server using HTTPS, preventing downgrade attacks.
+
+5. **Consider TLS 1.3:**
+   - If possible, enable support for TLS 1.3, which uses more secure MAC algorithms by default.
+
+**Example configurations for common web servers:**
+
+1. **Apache:**
+```apache
+SSLProtocol             all -SSLv3 -TLSv1 -TLSv1.1
+SSLCipherSuite          ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384
+SSLHonorCipherOrder     off
+SSLSessionTickets       off
+```
+
+2. **Nginx:**
+```nginx
+ssl_protocols TLSv1.2 TLSv1.3;
+ssl_ciphers ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384;
+ssl_prefer_server_ciphers off;
+```
+
+3. **IIS:**
+```powershell
+New-Item 'HKLM:\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\TLS 1.2\Server' -Force
+New-ItemProperty -path 'HKLM:\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\TLS 1.2\Server' -name 'Enabled' -value '1' -PropertyType 'DWord' -Force
+New-ItemProperty -path 'HKLM:\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\TLS 1.2\Server' -name 'DisabledByDefault' -value 0 -PropertyType 'DWord' -Force
+```
```

Comment:
i think its better to put like these: 

![image](https://github.com/user-attachments/assets/5a913ef9-9246-44e0-b73e-11ba83b17f76)


and please test in in a ticket comment to see if the rendering is correct.

Reply:
Done. Thanks.